src/main/java/com/simplesteph/kafka/GitHubSourceConnector.java

```diff
@@ -30,10 +30,25 @@ public void start(Map<String, String> map) {
     }
 
     @Override
-    public List<Map<String, String>> taskConfigs(int i) {
+    public List<Map<String, String>> taskConfigs(int n) {
+        List<GithubSourceTaskConfig> taskConfigs = config.getTasksConfig();
+
+        Integer maxRepositories = Math.min(n, taskConfigs.size());
+
+        if (maxRepositories < taskConfigs.size())
+            log.warn("Less workers than repositories, only first %d repositories will be extracted", maxRepositories);
```

Comment:
one task should definitely be able to extract multiple repo. Just create as many tasks as repos and let the framework take care of distributing that to the connect workers.